netlogo-core/src/main/api/Task.scala

```diff
@@ -2,10 +2,58 @@
 
 package org.nlogo.api
 
-sealed trait Task
+import org.nlogo.core.Syntax
+
+sealed trait Task {
+  /** Used to specify the number of arguments required to run a task.
+    *
+    * Note that a task may be supplied with more arguments depending
+    * on the number of arguments supplied to run/runresult.
+    * This only used to calculate argument count at the moment.
+    * Note that tasks as created by `task` are variadic in the number of arguments
+    * they accept. For instance, the task returned by `task [?1 + ?2]`
+    * can be run with 10 arguments and the last 8 will be ignored.
+    * The current versions only makes assertions that the number of arguments is
+    * greater than the syntax minimum.
+    *
+    * Future versions ''may'' make greater use of the information provided by
+    * `syntax`. Primitives returning tasks will want to ensure the task's syntax is
+    * not more restrictive than the expected argument(s).
+    */
+  def syntax: Syntax
```

Comment:
So that would need to go on the `ReporterTask` trait. It would work really well for scala extensions, but I think would be confusing for java extensions.